probe-rs/src/probe.rs

```diff
@@ -99,101 +99,110 @@ impl fmt::Display for BatchCommand {
     Registry(#[from] RegistryError),
-    /// The debug probe does not support the speed that was chosen.
-    /// Try to alter the selected speed.
-    #[error("The requested speed setting ({0} kHz) is not supported by the probe")]
+
+    /// The probe does not support he requested speed setting ({0} kHz).
     UnsupportedSpeed(u32),
+
+    /// You need to be attached to the target to perform this action.
+    ///
     /// The debug probe did not yet perform the init sequence.
     /// Try calling [`DebugProbe::attach`] before trying again.
-    #[error("You need to be attached to the target to perform this action")]
+    #[ignore_extra_doc_attributes]
     NotAttached,
+
+    /// You need to be detached from the target to perform this action.
+    ///
     /// The debug probe already performed the init sequence.
     /// Try running the failing command before [`DebugProbe::attach`].
-    #[error("You need to be detached from the target to perform this action")]
+    #[ignore_extra_doc_attributes]
     Attached,
-    /// Performing the init sequence on the target failed.
-    /// Check the wiring before continuing.
-    #[error("Failed to find the target or attach to the target")]
+
+    /// Failed to find or attach to the target. Please check the wiring before retrying.
     TargetNotFound,
+
+    /// Error in previous batched command.
+    BatchError(BatchCommand),
+
+    /// The '{function_name}' functionality is not implemented yet.
+    ///
     /// The variant of the function you called is not yet implemented.
     /// This can happen if some debug probe has some unimplemented functionality for a specific protocol or architecture.
-    #[error("Some functionality was not implemented yet: {0}")]
-    NotImplemented(&'static str),
-    /// The called debug sequence is not supported on given probe.
-    /// This is most likely happening because you are using an ST-Link, which are severely limited in functionality.
-    /// If possible, try using another probe.
-    #[error("This debug sequence is not supported on the used probe: {0}")]
-    DebugSequenceNotSupported(&'static str),
-    /// An error occurred during the previously batched command.
-    #[error("Error in previous batched command")]
-    BatchError(BatchCommand),
-    /// The used functionality is not supported by the selected probe.
+    #[ignore_extra_doc_attributes]
+    NotImplemented {
+        /// The name of the unsupported functionality.
+        function_name: &'static str,
+    },
+
+    /// The '{command_name}' functionality is not supported by the selected probe.
     /// This can happen when a probe does not allow for setting speed manually for example.
-    #[error("Command not supported by probe: {0}")]
-    CommandNotSupportedByProbe(&'static str),
+    CommandNotSupportedByProbe {
+        /// The name of the unsupported command.
+        command_name: &'static str,
+    },
+
     /// Some other error occurred.
-    #[error(transparent)]
+    #[display("{0}")]
```

Comment:
This will make it impossible to print the entire stack, doesn't it?

Reply:
Why?

Reply:
Ah, anyhow does this properly. Fair. Why did you change this tho? I don't get what the advantage is?

Reply:
The advantage is things working, you can't have two separate Display implementations for the same enum.